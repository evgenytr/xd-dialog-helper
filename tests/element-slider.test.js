const slider = require('../dialog-helper').types.SLIDER;
const actions = {
    change: jest.fn(() => undefined),
    values: jest.fn(() => {}),
    close: jest.fn(() => {}),
    cancel: jest.fn(() => {}),
    registerElement: jest.fn(() => undefined)
};

const rendered = slider.render('dialog', {
    id: 'slider',
    label: 'Move right',
    value: 5,
    htmlAttributes: {
        max: 10,
        min: 1
    },
    unit: 'cm'
}, actions);

global.console.error = jest.fn(() => {});

describe('Slider Element', () => {
    describe('render()', () => {
        it('should render correctly', () => {
            expect(rendered).toMatchSnapshot();
        });

        it('should log an error to the console for missing configuration', () => {
            expect(slider.render('dialog', {
                id: 'slider',
                label: 'Hello'
            }, actions)).toBeNull();
            expect(global.console.error).toBeCalled();
        });
    });
    describe('value()', () => {
        it('should return the correct numeric value', () => {
            expect(slider.value(rendered)).toBe(5);
        });
    });
    describe('validate()', () => {
        it('should always return true', () => {
            expect(slider.valid(rendered)).toBeTruthy();
        });
    });
    describe('name', () => {
        expect(slider.type).toMatchSnapshot();
    });
});
