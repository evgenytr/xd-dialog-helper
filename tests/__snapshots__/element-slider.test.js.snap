// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"Slider"`;

exports[`Slider Element render() should render correctly 1`] = `
Object {
  "input": <input
    id="dialog-slider"
    max="10"
    min="1"
    type="range"
  />,
  "props": Object {
    "htmlAttributes": Object {
      "max": 10,
      "min": 1,
    },
    "id": "slider",
    "label": "Move right",
    "unit": "cm",
    "value": 5,
  },
  "type": Object {
    "render": [Function],
    "type": "Slider",
    "valid": [Function],
    "value": [Function],
  },
  "wrapper": <label
    id="dialog-slider-wrapper"
  >
    <div
      class="row"
      style="justify-content: space-between;"
    >
      <span
        id="dialog-slider-label"
      >
        Move right
      </span>
      <span
        id="dialog-slider-value-label"
      >
        5cm
      </span>
    </div>
    <input
      id="dialog-slider"
      max="10"
      min="1"
      type="range"
    />
  </label>,
}
`;
